### ISSUE TITLE
AutoMapping Enumeration Class

### ISSUE THREAD

**Issue by TDK1964** [AutoMapper/AutoMapper#2322]
I'm experimenting with using enumeration classes [(link)](https://github.com/HeadspringLabs/Enumeration) instead and enums and "lookup tables". 

I have this scenario which displays a simple list in my view where I want to display the TaskStatus name from the enumeration class instead of the StatusId but I get this error "InvalidOperationException: The entity type 'TaskStatus' requires a primary key to be defined."

Is my approach all wrong?
```
<table clss="table">
    @foreach(var item in Model)
    {
        <tr>
            <td>@item.Id</td>
            <td>@item.Name</td>
            <td>@item.Status</td>
        </tr>
    }
</table>

public class Tasks : BaseEntity
{
	public string Name { get; set; }
	public DateTime? StartDate { get; set; }
	public DateTime? DueDate { get; set; }
	public byte StatusId { get; set; }
	public string AssignedTo { get; set; }

	public virtual TaskStatus Status { get; set; }
}
	
public class IndexVm
{
	public int Id { get; set; }
	public string Name { get; set; }
	public DateTime? StartDate { get; set; }
	public DateTime? DueDate { get; set; }
	public byte StatusId { get; set; }
	
	public TaskStatus Status { get; set; } 
}

public class TaskStatus : Enumeration<TaskStatus, int>
{
	public static readonly TaskStatus NotStarted           = new TaskStatus(1, "Not Started");
	public static readonly TaskStatus InProgress           = new TaskStatus(2, "In Progress");
	public static readonly TaskStatus Completed            = new TaskStatus(3, "Completed");
	public static readonly TaskStatus WaitingOnSomeoneElse = new TaskStatus(4, "Waiting on someone else");
	public static readonly TaskStatus Deferred             = new TaskStatus(5, "Deferred");
	
	private TaskStatus(int value, string displayName) : base(value, displayName) { }
}

public class MappingProfile : Profile
{
	public MappingProfile()
	{
		CreateMap<Tasks, IndexVm>()
			.ForMember(vm => vm.Status, conf => conf.MapFrom(ol => ol.Status.DisplayName));
	}
}



```

**Comment by lbargaoanu** [comment]
I think this is better suited for StackOverfl

**Comment by lbargaoanu** [comment]
I think this is better suited for StackOveflow.

**Comment by TDK1964** [comment]
Hmm helpful!
